Add check_lowerbound config for AOTI lowering

ColinPeppler opened the pull request:
Summary:
## Why:
PT has a 0/1 specialization assumption. Thus, unless backed_size_oblivious is on, the greatest lowerbound is [2+, ...]. So, this trips up a lot of models who can have a dynamic size with 0 or 1.

I want a way for AOTI lowering users to easily check their dynamic shape spec is correct via `AOTI_RUNTIME_CHECK_INPUTS=1` which is a runtime env var. In-order to do that, I need to avoid any errors triggered by a wrong lowerbound.

## What does it look like?
```
    // this is a lowerbound check
    if (arg0_1_size[0] < 2) {
        std::stringstream ss;
        ss << "input_handles[0]: dim value is too small at 0, "
           << "expected it to be >= 2, " << "but got: "
           << arg0_1_size[0] << "\n";
        throw std::runtime_error(ss.str());
    }
```

Differential Revision: D88203028




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

desertfire commented on torch/_inductor/codegen/cpp_wrapper_cpu.py:
```diff
@@ -423,7 +423,9 @@ def gen_check(handle_kind, idx, name, tensor):
-                    if not math.isinf(sym_range.lower):
+                    if config.aot_inductor.check_lowerbound and not math.isinf(
+                        sym_range.lower
+                    ):
```
Does it make sense to relax the lower bound to 0 if `sym_range.lower` is 2? Using a config here still can trip many users.

ColinPeppler committed:
Add check_lowerbound config for AOTI lowering (#169430)

Summary:

## Why:
PT has a 0/1 specialization assumption. Thus, unless backed_size_oblivious is on, the greatest lowerbound is [2+, ...]. So, this trips up a lot of models who can have a dynamic size with 0 or 1.

I want a way for AOTI lowering users to easily check their dynamic shape spec is correct via `AOTI_RUNTIME_CHECK_INPUTS=1` which is a runtime env var. In-order to do that, I need to avoid any errors triggered by a wrong lowerbound.

## What does it look like?
```
    // this is a lowerbound check
    if (arg0_1_size[0] < 2) {
        std::stringstream ss;
        ss << "input_handles[0]: dim value is too small at 0, "
           << "expected it to be >= 2, " << "but got: "
           << arg0_1_size[0] << "\n";
        throw std::runtime_error(ss.str());
    }
```

Reviewed By: cp2923, sevenEng

Differential Revision: D88203028
